osu.Desktop/Updater/AppImageUpdateManager.cs

```diff
@@ -0,0 +1,564 @@
+                        arguments = "--help";
+                        break;
+                    default:
+                        argument_spacer = "";
+                        break;
+                }
+
+                ProcessStartInfo info = new ProcessStartInfo
+                {
+                    FileName = Command,
+                    Arguments = $"{arguments}{argument_spacer}\"{AppImagePath}\"",
+                    UseShellExecute = false,
+                    RedirectStandardError = true,
+                    RedirectStandardOutput = true,
+                };
+
+                return info;
+            }
+            public enum States
+            {
+                NOUPDATESAVAILABLE,
+                UPDATESAVAILABLE,
+                DOWNLOADING,
+                VERIFYING,
+                COMPLETED,
+                CANCELLED
+            }
+            private static AppImageUpdateTool instance;
+            public static AppImageUpdateTool Instance
+            {
+                get
+                {
+                    if (instance is default(AppImageUpdateTool))
+                    {
+                        instance = new AppImageUpdateTool();
+                    }
+                    return instance;
+                }
+            }
+            private States state;
+            public static States State
+            {
+                get
+                {
+                    if (Instance.state is default(States))
+                    {
+                        Instance.state = Instance.hasUpdates() ?
+                            States.UPDATESAVAILABLE :
+                            States.NOUPDATESAVAILABLE;
+                    }
+                    return Instance.state;
+                }
+                private set
+                {
+                    Instance.state = value;
+                }
+            }
+            /// <summary>
+            /// Absolute path to AppImage file (with symlinks resolved)<para />
+            /// See: https://docs.appimage.org/packaging-guide/environment-variables.html
+            /// </summary>
+            public static string AppImagePath
+            {
+                get
+                {
+                    return IsAppImage ?
+                            // TODO: verify whether environment variable APPIMAGE is set in the deployed build
+                            Environment.GetEnvironmentVariable("APPIMAGE") :
+                            //Assume osu.AppImage otherwise (use this for debugging a test image)
+                            $"{RuntimeInfo.StartupDirectory}osu.AppImage";
+                }
+            }
+            /// <summary>
+            /// Checks if osu was launched as an AppImage
+            /// </summary>
+            public static bool IsAppImage
+            {
+                get
+                {
+                    return Environment.GetEnvironmentVariable("APPIMAGE") is string;
```

Comment:
Use pattern-matching when the returned string is used (same applies to all usages of this pattern).
```suggestion
                    return Environment.GetEnvironmentVariable("APPIMAGE") != null;
```